inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java

```diff
@@ -0,0 +1,555 @@
+        InlongClusterEntity mqCluster = mqClusters.get(0);
+        String strMqExtTag = mqCluster.getExtTag();
+        strMqExtTag = StringUtils.trimToEmpty(strMqExtTag);
+        Map<String, String> mqExtTagMap = new HashMap<>(MAP_SPLITTER.split(strMqExtTag));
+        // update extTag
+        if (isAppendTag) {
+            // append DataProxy extTag to MessageQueue extTag
+            proxyExtTagMap.forEach((k, v) -> mqExtTagMap.put(k, v));
+        } else {
+            // remove DataProxy extTag from MessageQueue extTag
+            proxyExtTagMap.forEach((k, v) -> mqExtTagMap.remove(k));
+        }
+        // update MessageQueue
+        String newExtTag = MAP_JOINER.join(mqExtTagMap);
+        mqCluster.setExtTag(newExtTag);
+        clusterMapper.updateById(mqCluster);
+        return null;
+    }
+
+    /**
+     * Synchronize all topic from cluster tag to message queue cluster
+     */
+    public String synchronizeTopic(MessageQueueSynchronizeTopicRequest request) {
+        String mqClusterName = request.getName();
+        // check parameters
+        if (StringUtils.isEmpty(mqClusterName)) {
+            return "miss message queue cluster name.";
+        }
+        // find cluster
+        List<InlongClusterEntity> mqClusters = clusterMapper.selectByKey(null, mqClusterName, null);
+        if (mqClusters.size() <= 0) {
+            return "Can not find message queue cluster by name:" + mqClusterName;
+        }
+        if (mqClusters.size() > 1) {
+            return String.format("MessageQueue cluster:%s data is more than 1.", mqClusterName);
+        }
+        // find cluster tag
+        InlongClusterEntity mqCluster = mqClusters.get(0);
+        String clusterTag = mqCluster.getClusterTags();
+        if (StringUtils.isEmpty(clusterTag)) {
+            return String.format("Cluster tag of message queue cluster:%s is null.", mqClusterName);
+        }
+        // find group entities
+        InlongGroupPageRequest groupRequest = new InlongGroupPageRequest();
+        groupRequest.setIsAdminRole(true);
+        groupRequest.setClusterTagList(Collections.singletonList(clusterTag));
+        List<InlongGroupEntity> groupEntities = groupMapper.selectByCondition(groupRequest);
+        String mqType = mqCluster.getType();
+        if (StringUtils.equalsIgnoreCase(mqType, MQType.PULSAR)) {
+            return this.createPulsarTopic(mqCluster, groupEntities);
+        } else if (StringUtils.equalsIgnoreCase(mqType, MQType.KAFKA)) {
+            return this.createKafkaTopic(mqCluster, groupEntities);
+        } else if (StringUtils.equalsAnyIgnoreCase(mqType, MQType.TUBEMQ)) {
+            return this.createTubeTopic(mqCluster, groupEntities);
+        } else {
+            return String.format("Unknown message queue type:%s", mqType);
+        }
+    }
+
+    /**
+     * createPulsarTopic
+     */
+    private String createPulsarTopic(InlongClusterEntity cluster, List<InlongGroupEntity> groupInfos) {
```

Comment:
Very famaliar logic with the **_createXXXXTopic_** in **ClusterTagService**. 
Can use **ClusterTagService** dicretly?

Reply:
It is different, ClusterTagService create a topic for every MessageQueue cluster.
MessageQueueService create all topic of this cluster tag.